### DISCUSSION TITLE
How to import snowflakehook and snowflakeoperator

### DISCUSSION THREAD

**Discussion by Nupurgopali** [apache/airflow#25191] [General]
To send csv files to snowflake via airflow, I am using snowflakehook and snowflakeoperator.

I am importing the libraries like this:

from airflow.contrib.hooks.snowflake_hook import SnowflakeHook
from airflow.contrib.operators.snowflake_operator import SnowflakeOperator

I am running the airflow using docker(on windows system) and am facing this error: 'ModuleNotFoundError: No module named 'snowflake'. Not able to understand what is the issue.

When I tried to import them via airflow.providers class, I get another error : 'ModuleNotFoundError: No module named 'airflow.providers''.

Ps: I have already installed apache-airflow-providers-snowflake

Do let me know if you have faced this error before and how you rectified it. Thanks in advance!

**Comment by potiuk** [comment]
You likely have a broken installation and more than one version of airlfow/providers install (my quess you have a weird mixture of Airflow 1.10 and Airflow 2. I am afraid the problems you have indicate, that very basic foundation of your Airflow installation are broken and you need to start from scratch.

You need to understand and check which Python modules you have and use, which version you have how you build and install airlfow. I thin am afraid you have some bad combination of packages installed. There is not much help we can offer besides you looking at what you have installed and what is broken. 

If you want to manage your installation you need to learn about pip, installation, packages, and how to check if they are installed and which versions. I can give you few starting points:

* learn about `pip`
* learn about `pip freeze`
* learn about `python -m site` and where python packages are installed
* learn about building docker image enough to be able to rebuild yours and only install what is needed. Your current image is likely broken - you should more likely than not start with airflow reference image https://airflow.apache.org/docs/docker-stack/index.html and re-create your image from scratch.
* look at airflow deprecation policies/changelog and warnings you will get. The "airflow.contrib" is deprecated and it should raise exceptions (unless how I suspect you have weird combination of Ariflow 1 and 2 installed together in two different place - then contrib might not raise deprecation.

You can find a lot of information about those when you google.

**Comment by mkalinna** [comment]
The modules can be loaded with
`from airflow.providers.snowflake.operators.snowflake import SnowflakeOperator`
`from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook`
Example DAGs are linked in the documentation and can be found here:
https://github.com/apache/airflow/tree/providers-snowflake/4.2.0/tests/system/providers/snowflake

Hope that helps.
